oci/table_oci_core_nat_gateway.go

```diff
@@ -0,0 +1,224 @@
+				Description: "Specifies whether the NAT gateway blocks traffic through it.",
+				Type:        proto.ColumnType_BOOL,
+			},
+
+			// tags
+			{
+				Name:        "defined_tags",
+				Description: ColumnDescriptionDefinedTags,
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "freeform_tags",
+				Description: ColumnDescriptionFreefromTags,
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// Standard Steampipe columns
+			{
+				Name:        "tags",
+				Description: ColumnDescriptionTags,
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.From(natGatewayTags),
+			},
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("DisplayName"),
+			},
+
+			// Standard OCI columns
+			{
+				Name:        "compartment_id",
+				Description: ColumnDescriptionCompartment,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("CompartmentId"),
+			},
+			{
+				Name:        "tenant_id",
+				Description: ColumnDescriptionTenant,
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getTenantId,
+				Transform:   transform.FromValue(),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listCoreNatGateway(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	region := plugin.GetMatrixItem(ctx)[matrixKeyRegion].(string)
+	compartment := plugin.GetMatrixItem(ctx)[matrixKeyCompartment].(string)
+	logger.Debug("oci.listCoreNatGateway", "Compartment", compartment, "OCI_REGION", region)
+
+	// Create Session
+	session, err := coreVirtualNetworkService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	request := core.ListNatGatewaysRequest{
+		CompartmentId: types.String(compartment),
+		RequestMetadata: oci_common.RequestMetadata{
+			RetryPolicy: getDefaultRetryPolicy(),
+		},
+	}
+
+	pagesLeft := true
+	for pagesLeft {
+		response, err := session.VirtualNetworkClient.ListNatGateways(ctx, request)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, natGateways := range response.Items {
+			d.StreamListItem(ctx, natGateways)
+		}
+		if response.OpcNextPage != nil {
+			request.Page = response.OpcNextPage
+		} else {
+			pagesLeft = false
+		}
+	}
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTION
+
+func getNatGateway(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getUser")
+	logger := plugin.Logger(ctx)
+	region := plugin.GetMatrixItem(ctx)[matrixKeyRegion].(string)
+	compartment := plugin.GetMatrixItem(ctx)[matrixKeyCompartment].(string)
+	logger.Debug("oci.getNatGateway", "Compartment", compartment, "OCI_REGION", region)
+
+	// Rstrict the api call to only root compartment/ per region
+	if !strings.HasPrefix(compartment, "ocid1.tenancy.oc1") {
+		return nil, nil
+	}
+
+	id := d.KeyColumnQuals["id"].GetStringValue()
+
```

Comment:
`> select * from oci_new.oci_core_nat_gateway where id = ' '
Error: 2 list calls returned errors:
 Service error:InvalidParameter. invalid natgateway OCID:  . http status code: 400. Opc request id: 30603e33f21f2b9c06a4635eccae2f02/DB9FF17679A97176DF1F8FBE58407926/1ED19160D2218D6585D063D51D166DB6
Service error:InvalidParameter. invalid natgateway OCID:  . http status code: 400. Opc request id: 8f4c28b732f72caed92c24d1a042afa5/BAFA61607386F94018703FA51550BA89/96B446CC4484F430EC6A69E861A51DD0
> select * from oci_new.oci_core_nat_gateway where id = ''
Error: 2 list calls returned errors:
 value cannot be empty for field NatGatewayId in path`
 
 Handle both error for get call.